ArtOfIllusion/src/artofillusion/ScrollViewTool.java

```diff
@@ -0,0 +1,301 @@
+/* Copyright (C) 2017 by Petri Ihalainen
```

Comment:
One reason to put scroll wheel handling into it's own file/class was, that the ViewerCanvas is not aware of the parent window that it is attached to. For a while I had the scroll events handled in the VC and MoveViewTool handle them a bit more to create the view cone graphics to the other views but obviously that was not a very logical to do it.

And it is not a "tool" in the sense either, that I did not see any reason to make it an extension to EditingTool. _(IMHO the Move and Rotate tools are not really "editing tools" either as don't edit anything. Though that arrangement makes sense in the way, that they are assigned to Icons.)_

By click-only tools you mean that they have icons in the tool palette? And they would only produce the very basic functions? -- Anyway the idea of a mose-handler class sounds about right to me. Somehow I'm thinking that the ViewerCanvas is large enough a file already and adding more weight to it .. hmmm...

On the other hand most places where the ViewerCanvas is used would still be using all the mouse-event handling -- Maybe the only exception being the UVMappingViewer, that would do just fine with less options.